src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,32 @@
 import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from '../../app/store';
+import { filterActions } from '../../features/filter';
+import { Status } from '../../types/Status';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const query = useSelector((state: RootState) => state.filter.query);
```

Comment:
did you know how to create useAppSelector and useAppDispatch hooks? better to use them and remove types everywhere
```suggestion
  const query = useAppSelector(state => state.filter.query);
```